### ISSUE TITLE
Space in one variable breaks others

### ISSUE THREAD

**Issue by brad** [ansible/ansible#6329]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.5
##### Environment:

Arch Linux managing Centos 6
##### Summary:

When I accidentally use a space instead of an underscore, all the variables in the command are ignored and each character is treated literally.
##### Steps To Reproduce:

``` YAML

---

- hosts: all
  vars:
    - test: "test1"
    - test_var: "test2"

  tasks:
    - command: echo {{ test }}{{ test var }}
```
##### Expected Results:

I think there are two possible results that could improve things:
1. The play errors out, explaining that there should not be spaces in variables
2. The play continues and evaluates all but the mucked up variable correctly. Something like this: `TASK: [command echo test{{test var}}]`

Either way would make it easy for me to see the problem. As it is, it took me a while to notice what the problem was (mostly because the line was on the last line of the editor so it wasn't easy to see that it was a space instead of an underscore).
##### Actual Results:

``` bash
[default] Running provisioner: ansible...

PLAY [vagrant] **************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [192.168.111.222]

TASK: [command echo {{test}}{{test var}}] ************************************* 
changed: [192.168.111.222]

PLAY RECAP ******************************************************************** 
192.168.111.222            : ok=2    changed=1    unreachable=0    failed=0
```

**Comment by jimi-c** [comment]
@brad, could you try the patch above and let me know if it resolves the issue for you? Thanks!

**Comment by brad** [comment]
@jimi-c Excellent, thank you! With the patch, I get this output:

```
[default] Running provisioner: ansible...

PLAY [vagrant] **************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [192.168.111.222]
ERROR: there was an error in the template: command echo {{test}}{{test var}}
Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
```

Seems like much more sensible behavior. :beers:

**Comment by jimi-c** [comment]
Great, merged that in. Thanks!

**Comment by brad** [comment]
Wow, that was easy. Thanks for being so easy to work with!

**Comment by jimi-c** [comment]
The above patch was causing issues and was reverted, so re-opening this.

**Comment by brad** [comment]
What issues did the patch cause? I'm struggling (and failing) to find the history of that.

**Comment by jimi-c** [comment]
@brad it broke certain non-jinja variables, like shell variables and other things that might contain a set of braces.

**Comment by brad** [comment]
Ahh ok. Good to know.

**Comment by jimi-c** [comment]
Looking back at this, it appears to be fixed now in the devel (v2/ansible 2.0) code base.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
